sled-agent/src/sim/config.rs

```diff
@@ -60,12 +70,55 @@ pub struct Config {
     pub nexus_address: SocketAddr,
     /// configuration for the sled agent dropshot server
     pub dropshot: ConfigDropshot,
-    /// configuration for the sled agent debug log
-    pub log: ConfigLogging,
     /// configuration for the sled agent's storage
     pub storage: ConfigStorage,
     /// configuration for the sled agent's updates
     pub updates: ConfigUpdates,
     /// configuration to emulate the sled agent's hardware
     pub hardware: ConfigHardware,
 }
+
+impl Config {
+    pub fn for_testing(
```

Comment:
This commonizes boilerplate from several callers (nexus/test-util, the simulated sled agent executable, and a new caller in the collector tests).

Reply:
Awesome!